### PR TITLE
Module documentation fixes and updates

### PR DESCRIPTION
**Pull request by cbrnrd**
I did a quick run through of some module docs and fixed some minor things. This is _not_ all of the problems in the docs. I'm sure there are more. 
This PR is a few things:
1. It fixes some grammatical errors (commas, spelling and the like)
2. Adds 'Introduction' header to particularly long starting descriptions 
    - I added this usually if the description was >~ 5 lines
3. Added clarifying statements (see censys_search.md)
4. Fix some broken links
5. Added some additional information (see archmigrate.md)
6. Fixed some minor but annoying markdown header typos
    - ##they looked like this
7. Removed some eyesores
    - See ms17_010_eternalblue, hashdump, and archmigrate

Paging @h00die because he's the docmaster, but if anyone else has some suggestions or additional edits, please tell me!

### CODE REVIEW

File: documentation/modules/exploit/multi/local/allwinner_backdoor.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
+  ## Introduction
```
Comment (h00die) [inline documentation/modules/exploit/multi/local/allwinner_backdoor.md]: add a new line after intro and remove the spaces before the pounds
The lines after it don't look so hot in MD format either, mind taking a stab at cleaning them up (this was my module WAY back in the day, my how things have changed)

File: documentation/modules/exploit/windows/fileformat/cve_2017_8464_lnk_rce.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ A fix was released in the June 2017 Patch Tuesday.
 
 ## Vulnerable Setup
 
-To set up the vulnerable environment, install a Windows version without the patch for CVE-2017-8464. To test the bypass, ensure that MS10-046 & MS15-020 are installed.
+To set up the vulnerable environment, install a Windows version without the patch for CVE-2017-8464. To test the bypass, ensure that MS10-046 & MS15-020 are exploitable.
```
Comment (h00die) [inline documentation/modules/exploit/windows/fileformat/cve_2017_8464_lnk_rce.md]: this sentence doesn't make much sense to me in general. There are no prior mentions in the docs to ms10-046 and a 'bypass' is only mentioned in the module info.  I think more info needs to be added to what we're bypassing, and why ms10 is relevant here.

File: documentation/modules/post/windows/manage/archmigrate.md [hunk 3]
```diff
@@ -12,13 +12,25 @@ This module was not tested against, but may work against:
 
   1. Other versions of Windows that are x64.
 
-## Verification Steps
+## Options
+
+- #### EXE
```
Comment (h00die) [inline documentation/modules/post/windows/manage/archmigrate.md]: These don't follow the normal flow, should be **EXE**

File: documentation/modules/post/windows/manage/archmigrate.md [hunk 4]
```diff
@@ -12,13 +12,25 @@ This module was not tested against, but may work against:
...
+    The executable to start and migrate into. Default: `C:\windows\sysnative\svchost.exe`
+
+- #### FALLBACK
```
Comment (h00die) [inline documentation/modules/post/windows/manage/archmigrate.md]: These don't follow the normal flow, should be **FALLBACK**

File: documentation/modules/post/windows/manage/archmigrate.md [hunk 5]
```diff
@@ -12,13 +12,25 @@ This module was not tested against, but may work against:
...
+    If the selected migration executable does not exist, fallback to a sysnative file. Default: `true`
+
+- #### IGNORE_SYSTEM
```
Comment (h00die) [inline documentation/modules/post/windows/manage/archmigrate.md]: These don't follow the normal flow, should be **IGNORE_SYSTEM**

**Review by h00die — CHANGES_REQUESTED** [review]

**Comment by h00die** [comment]
overall looks good, nice changes including on 2 of my own docs :).  I'll land soon as you get those minor changes made, thanks a bunch for this!!!!!

File: documentation/modules/auxiliary/admin/smb/ms17_010_command.md [hunk 6]
```diff
@@ -1,3 +1,5 @@
+## Introduction
+
 MS17-010 are psexec are two of the most popular exploits against Microsoft Windows. This module bolts the two together.
```
Comment (bcoles) [inline documentation/modules/auxiliary/admin/smb/ms17_010_command.md]: "MS17-010 and psexec"  ?

File: documentation/modules/auxiliary/gather/shodan_honeyscore.md [hunk 7]
```diff
@@ -1,3 +1,4 @@
+## Introduction
 The `shodan_honeyscore` module utilizes the [Shodan](https://www.shodan.io/) API to determine whether or not a server is a honeypot or not.
```
Comment (bcoles) [inline documentation/modules/auxiliary/gather/shodan_honeyscore.md]: `whether or not a server is a honeypot or not`

Repeated `or not`. Second instance is redundant.

**Comment by cbrnrd** [comment]
Thanks for the review :)  I'll change these by tomorrow morning.

File: documentation/modules/exploit/windows/fileformat/office_word_hta.md [hunk 8]
```diff
@@ -1,11 +1,13 @@
+## Introduction
+
 Microsoft Office is an office suite of applications, servers, and services developed by Microsoft. Microsoft Office contains Microsoft Word, Microsoft Excel, Microsoft PowerPoint and so on. They can support OLE data integration and Virtusl Basic for Application scripting langauage.
```
Comment (bcoles) [inline documentation/modules/exploit/windows/fileformat/office_word_hta.md]: `Virtusl Basic` ? Should be `Visual Basic` ?

`langauage` should be `language`

File: documentation/modules/exploit/windows/fileformat/cve_2017_8464_lnk_rce.md (hunk 2 above)
Comment (cbrnrd) [reply documentation/modules/exploit/windows/fileformat/cve_2017_8464_lnk_rce.md]: Based on [this](https://www.cvedetails.com/cve/cve-2017-8464) and [this](https://www.rapid7.com/db/vulnerabilities/msft-cve-2017-8464), you aren't bypassing anything. Should the sentence just be removed?

Comment (h00die) [reply documentation/modules/exploit/windows/fileformat/cve_2017_8464_lnk_rce.md]: https://www.rapid7.com/db/modules/exploit/windows/fileformat/cve_2017_8464_lnk_rce mentions bypassing a CPL (control panel?) Whitelist

Comment (h00die) [reply documentation/modules/exploit/windows/fileformat/cve_2017_8464_lnk_rce.md]: @zeroSteiner any insight here?

**Review by h00die — APPROVED** [review]

File: documentation/modules/exploit/windows/fileformat/cve_2017_8464_lnk_rce.md (hunk 2 above)
Comment (zeroSteiner) [reply documentation/modules/exploit/windows/fileformat/cve_2017_8464_lnk_rce.md]: Yeah, my understanding was that setting the folder ID in the `SpecialFolderDataBlock` to the Control Panel was causing the white list to be bypassed allowing an arbitrary CPL, (in the case of the module a DLL) to be loaded.

I don't see any reference to MS10-046 in either the fileformat or local variants of the module. IIRC, I didn't write the documentation for either of those so I'm not entirely sure how MS10-046 relates to them.

**Comment by cbrnrd** [comment]
I removed the sentence we were having trouble with. Any other changes?

**Comment by h00die** [comment]
I'm good with this, it can always be made better later if someone had something else.  At this point its a bunch of good changes.land coming shortly!

**Comment by h00die** [comment]
# Release Notes
Updates to some spelling and grammar on docs, as well as consistency upgrades.

**Comment by h00die** [comment]
@cbrnrd as always, thanks for the touch ups and changes.  Docs may not be sexy, but when something goes wrong and there aren't any its super aggravating!

**Pull request merged**